pkg/dmaasbackup/nonperiodic_backup.go

```diff
@@ -14,36 +14,20 @@ limitations under the License.
 package dmaasbackup
 
 import (
-	"github.com/mayadata-io/dmaas-operator/pkg/apis/mayadata.io/v1alpha1"
 	"github.com/pkg/errors"
+
+	"github.com/mayadata-io/dmaas-operator/pkg/apis/mayadata.io/v1alpha1"
 )
 
 func (d *dmaasBackup) processNonperiodicConfigSchedule(dbkp *v1alpha1.DMaaSBackup) error {
 	d.logger.Debug("Processing non fullbackup")
+	defer d.logger.Debug("Processing non-fullbackup completed")
 
 	// non full backup schedule can be of two types, schedule or normal backup
-	// check for scheduled backup
-	if dbkp.Spec.VeleroScheduleSpec.Schedule != "" {
-		// dmaasbackup has schedule spec
-		// create schedule if not created
-		if len(dbkp.Status.VeleroSchedules) == 0 {
-			schedule, err := d.createSchedule(dbkp)
-			if err != nil {
-				d.logger.WithError(err).
-					Errorf("failed to create schedule")
-				return errors.Wrapf(err, "failed to create schedule")
-			}
-			// add schedule information to dmaasbackup
-			appendVeleroSchedule(dbkp, schedule)
-			d.logger.Infof("Schedule=%s created", schedule.Name)
-			return nil
-		}
-		// schedule is already created
-		return nil
-	}
-
+	// check for non-scheduled backup
 	// dmaasbackup have information for normal backup
-	if dbkp.Status.VeleroBackupName == nil {
+	if dbkp.Status.VeleroBackupName == nil &&
+		dbkp.Spec.VeleroScheduleSpec.Schedule == "" {
 		// need to create velero backup
 		bkp, err := d.createBackup(dbkp)
 		if err != nil {
```

Comment:
no need to error out if err is of type AlreadyExists

Reply:
ok